EventFilter/TwinMuxRawToDigi/src/TwinMuxFEDReader.cc

```diff
@@ -0,0 +1,688 @@
+//-------------------------------------------------
+//
+//   Class: DTTM7FEDReader
+//
+//   L1 DT TwinMux Raw-to-Digi
+//
+//
+//
+//   Author :
+//   C. F. Bedoya  - CIEMAT
+//   G. Codispoti -- INFN Bologna
+//   J. Pazzini   -- INFN Padova
+//
+//--------------------------------------------------
+
+#include "EventFilter/TwinMuxRawToDigi/interface/TwinMuxFEDReader.h"
+
+#include "DataFormats/Common/interface/Handle.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include <iostream>
+#include <fstream>
+
+DTTM7FEDReader::DTTM7FEDReader(const edm::ParameterSet& pset) :
+
+  debug_( pset.getUntrackedParameter<bool>("debug", false) ), 
+  passbc0_( pset.getUntrackedParameter<bool>("passbc0", false) ), 
+  nfeds_(0),
+  DTTM7InputTag_( pset.getParameter<edm::InputTag>("DTTM7_FED_Source") ),
+  feds_( pset.getUntrackedParameter<std::vector<int> >("feds", std::vector<int>()) ),
+  wheels_( pset.getUntrackedParameter<std::vector<int> >("wheels", std::vector<int>())),
+  amcsecmap_( pset.getUntrackedParameter<std::vector<long long int> >("amcsecmap", std::vector<long long int>()))
+
+{
+    
+  produces<L1MuDTChambPhContainer>();
+  produces<L1MuDTChambThContainer>();
+  Raw_token = consumes<FEDRawDataCollection> (DTTM7InputTag_);
+ 
+  nfeds_ = feds_.size();
+  
+  if ( nfeds_ != wheels_.size() )
+    throw cms::Exception("TwinMux_unpacker") << "Configuration file error. Size of \'wheels\' and \'feds\' differs.\n";  
+
+  if ( amcsecmap_.size() != wheels_.size() )
+    throw cms::Exception("TwinMux_unpacker") << "Configuration file error. Size of \'wheels\' and \'amcsecmap\' differs.\n";  
+  
+  for (size_t wh_i = 0; wh_i < amcsecmap_.size(); ++wh_i){
+    std::map< short, short > whmap;      
+    for (size_t amc_i = 1; amc_i < 13; ++amc_i ){
+      short shift = (12-amc_i)*4;
+      whmap[amc_i] = ( amcsecmap_.at(wh_i) >> shift ) & 0xF;
+    }
+    amcsec_.push_back(whmap);
```

Comment:
using a map to store keys from 1 to 12 is not appropriate. Please use and std::array<short, 12> instead. Or is something else done with this map elsewhere?